motr/io_nw_xfer.c

```diff
@@ -2135,13 +2136,17 @@ static int nw_xfer_req_dispatch(struct nw_xfer_request *xfer)
 		}
 		if (ti->ti_req_type == TI_COB_CREATE &&
 		    ioreq_sm_state(ioo) == IRS_WRITING) {
-			/*
-			 * An error returned by rpc post has been ignored.
-			 * It will be handled in the respective bottom half.
-			 */
 			M0_LOG(M0_DEBUG, "item="ITEM_FMT" osr_xid=%"PRIu64,
 				ITEM_ARG(item), item->ri_header.osr_xid);
 			rc = m0_rpc_post(item);
+			if (rc != 0 ) {
+				rpc_error = true;
+				M0_LOG(M0_WARN, "item="ITEM_FMT" osr_xid=%"
+				       PRIu64"rc = %d, ri_error = %d",
+				ITEM_ARG(item), item->ri_header.osr_xid,
+				rc, item->ri_error);
+				goto out;
+			}
```

Comment:
What if we only do this change as the comments say:
```
diff --git a/motr/io_nw_xfer.c b/motr/io_nw_xfer.c
index b26e628c4..76697c8e0 100644
--- a/motr/io_nw_xfer.c
+++ b/motr/io_nw_xfer.c
@@ -2142,6 +2142,7 @@ static int nw_xfer_req_dispatch(struct nw_xfer_request *xfer)
                        M0_LOG(M0_DEBUG, "item="ITEM_FMT" osr_xid=%"PRIu64,
                                ITEM_ARG(item), item->ri_header.osr_xid);
                        rc = m0_rpc_post(item);
+                       rc = 0;/* Ignore error. See above comments. */
                        M0_CNT_INC(nr_dispatched);
                        m0_op_io_to_rpc_map(ioo, item);
                        continue;
@@ -2160,6 +2161,7 @@ static int nw_xfer_req_dispatch(struct nw_xfer_request *xfer)
                                                 ITEM_ARG(item),
                                                 item->ri_header.osr_xid);
                                rc = m0_rpc_post(item);
+                               rc = 0;/* Ignore error. See above comments. */
                                M0_CNT_INC(nr_dispatched);
                                m0_op_io_to_rpc_map(ioo, item);
                        }

```

Reply:
I think we need not want to continue sending more fops after detecting rpc error for one of the dispatch fops as look to be done for CREATE/TRUNCATE operation. Whereas we do come out for any error in dispatch of  io fop.

So what we have done currently is a common handling for create n io fops if rc != 0 where basically dispatching fop fails to the server .

Reply:
OK